rotkehlchen/externalapis/opensea.py

```diff
@@ -42,12 +45,45 @@
 log = RotkehlchenLogsAdapter(logger)
 
 
+def _read_floor_asset_price_from_collection(
+        floor_price_symbol: str,
+        collection_payment_tokens: list[dict[str, str]],
+) -> Asset:
+    """
+    Get the asset currency for the floor price reported by opensea using the list
+    of assets listed in the collection's payment tokens.
+
+    TODO: The api has a bug reported in the ticket #1455100 and the chain is reported
+    always as ethereum even if the asset is not an ethereum token.
+    When fixed adjust this function to use the chain.
+
+    May raise:
+    - UnknownAsset if the token wasn't found in the collection payment tokens.
+    """
+    for payment_token in collection_payment_tokens:
+        if payment_token['symbol'] == floor_price_symbol:
```

Comment:
This and the other key lookups can raise KeyError. You need to handle it or put it in the docstring this propagates upwards and handle it at the caller